app/models/concerns/type_of_means_assessment.rb

```diff
@@ -12,11 +12,21 @@ def requires_means_assessment?
     !evidence_of_passporting_means_forthcoming?
   end
 
-  def requires_full_means_assessment?
+  def requires_full_means_assessment? # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
     return false unless requires_means_assessment?
     return true if income_above_threshold? || has_frozen_assets?
+    return false if summary_only? || committal?
+    return true if income.client_owns_property == 'yes'
+    return true if income.has_savings == 'yes'
+    return false if no_property? && no_savings?
 
-    !summary_only? && !(no_property? && no_savings?)
+    raise Errors::CannotYetDetermineFullMeans
+  end
+
+  def extent_of_means_assessment_determined?
+    requires_full_means_assessment? || true
```

Comment:
Not sure I understand this. If `requires_full_means_assessment?` resolves to nil  then the default is `true`?

Reply:
Oh, right I see. If it gets to the end and hasn't returned something `requires_full_means_assessment?` throws an error which then skips this OR statement and hits the rescue